### ISSUE TITLE
forbid-component-props: How to distinguish a component from a library?

### ISSUE THREAD

**Issue by mnn** [yannickcr/eslint-plugin-react#2766]
Situation:

```tsx
import {Layout as AntdLayout} from 'antd';
...
            <AntdLayout.Content className={contentClassNames}>
...
```

Adding `AntdLayout` nor `AntdLayout.Content` into `allowedFor` isn't working. I don't want to add `Content`, because then linter won't report an error when our `Content` component is used with `className`.

Edit: I have attempted to improve it in https://github.com/yannickcr/eslint-plugin-react/pull/2767.

**Comment by ljharb** [comment]
I would expect `AntdLayout.Content` to work; this seems like an omission.
